build-android

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+# =======================
+# Preparetion
+# =======================
+# 1. set id, versionName and versionCode in <widget> of config.xml
+# 2 Download google-services.json to ~/Downloads
+# 3. Set AT API Subscription Key to the following variable
+AT_API_SUBSCRIPTION_KEY=XXXXXXXXXXXXX
+# 4. Download keystore file as ~/Downloads/1.keystore
+# 5. Set keystore password to the following variable
+ANDROID_KEY_STORE_PASSWORD=XXXXXXX
+# 6. Install nvm if you haven't
+export NVM_DIR="$HOME/.nvm"
+. "/usr/local/opt/nvm/nvm.sh"
+
+#========================
+# Prepare android sdk tools
+# =======================
+# Download android-tools-25.zip to ~/Downloads
+# mv ~/Library/Android/sdk/tools ~/Library/Android/sdk/tools-backup
+# unzip ~/Downloads/android-tools-25.zip -d ~/Library/Android/sdk/
+# mv ~/Library/Android/sdk/android-tools-25 ~/Library/Android/sdk/tools
+
+#========================
+# setup node 6
+# =======================
+nvm install 6.14.3
+nvm use v6
+
+#========================
+# setup auckland transport api key
+# =======================
+# please remember to replace place holder with the api key before running this script
+
+sed -ia "s|API_SUBSCRIPTION_ID|$API_SUBSCRIPTION_ID|g" ~/git/trackmybus/app/js/config.js
+
+#========================
+# clear all
+# =======================
+rm -fr app/lib platforms plugins www
+
+#========================
+# install packages
+# =======================
+npm install
+
+#========================
+# reinstall cordova plugins
+# =======================
+
+./node_modules/.bin/cordova plugin rm cordova-plugin-customurlscheme
```

Comment:
Instead of doing all of this, wouldn't it have been easier to change the `postinstall` script in `package.json` to avoid doing the` ionic state restore`? That way you could manually add each platform and install the plugin you want?

Or maybe you could have remove the caret (^) from the `package.json` and `config.xml` so you fix the exact verison of each plugin you want to install.